primitives/src/merkle_tree/namespaced_merkle_tree.rs

```diff
@@ -825,5 +839,35 @@ mod nmt_tests {
             .unwrap()
             .is_ok());
         assert_eq!(leaves, []);
+
+        // Check absence proof case when the namespace falls inside of the tree range
+        let absence_proof = tree.get_namespace_proof(3);
```

Comment:
I tried to compute an absence proof for ns=first_ns -1=0 but I get an error. Is this normal?

```rust
let absence_proof = tree.get_namespace_proof(first_ns-1);

attempt to subtract with overflow
thread 'merkle_tree::namespaced_merkle_tree::nmt_tests::test_nmt' panicked at 'attempt to subtract with overflow', primitives/src/merkle_tree/namespaced_merkle_tree.rs:655:62
stack backtrace:
   0: rust_begin_unwind
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
   1: core::panicking::panic_fmt
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
   2: core::panicking::panic
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:114:5
   3: <jf_primitives::merkle_tree::namespaced_merkle_tree::NMT<E,H,Arity,N,T> as jf_primitives::merkle_tree::namespaced_merkle_tree::NamespacedMerkleTreeScheme>::get_namespace_proof
             at ./src/merkle_tree/namespaced_merkle_tree.rs:655:62
   4: jf_primitives::merkle_tree::namespaced_merkle_tree::nmt_tests::test_nmt
             at ./src/merkle_tree/namespaced_merkle_tree.rs:831:29
   5: jf_primitives::merkle_tree::namespaced_merkle_tree::nmt_tests::test_nmt::{{closure}}
             at ./src/merkle_tree/namespaced_merkle_tree.rs:778:19
   6: core::ops::function::FnOnce::call_once
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Reply:
Anyways it would be good to have a test for an absence proof of a namespace smaller than `root.min_namespace`.

Reply:
Good catch - we need to ensure that `i > 0`

Reply:
@philippecamacho fixed